command/agent/agent_endpoint.go

```diff
@@ -740,3 +742,148 @@ func (s *HTTPServer) AgentHostRequest(resp http.ResponseWriter, req *http.Reques
 
 	return reply, rpcErr
 }
+
+// AgentSchedulerWorkerInfoRequest is used to query the running state of the
+// agent's scheduler workers.
+func (s *HTTPServer) AgentSchedulerWorkerInfoRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	srv := s.agent.Server()
+	if srv == nil {
+		return nil, CodedError(http.StatusBadRequest, ErrServerOnly)
+	}
+	if req.Method != http.MethodGet {
+		return nil, CodedError(http.StatusMethodNotAllowed, ErrInvalidMethod)
+	}
+
+	var secret string
+	s.parseToken(req, &secret)
+
+	// Check agent read permissions
+	if aclObj, err := s.agent.Server().ResolveToken(secret); err != nil {
+		return nil, CodedError(http.StatusInternalServerError, err.Error())
+	} else if aclObj != nil && !aclObj.AllowAgentRead() {
+		return nil, CodedError(http.StatusForbidden, structs.ErrPermissionDenied.Error())
+	}
+
+	schedulersInfo := srv.GetSchedulerWorkersInfo()
+	response := &agentSchedulerWorkersInfo{
+		ServerID:   srv.LocalMember().Name,
+		Schedulers: make([]agentSchedulerWorkerInfo, len(schedulersInfo)),
+	}
+
+	for i, workerInfo := range schedulersInfo {
+		response.Schedulers[i] = agentSchedulerWorkerInfo{
+			ID:                workerInfo.ID,
+			EnabledSchedulers: make([]string, len(workerInfo.EnabledSchedulers)),
+			Started:           workerInfo.Started.UTC().Format(time.RFC3339Nano),
+			Status:            workerInfo.Status,
+			WorkloadStatus:    workerInfo.WorkloadStatus,
+		}
+		copy(response.Schedulers[i].EnabledSchedulers, workerInfo.EnabledSchedulers)
+	}
+
+	return response, nil
+}
+
+// AgentSchedulerWorkerConfigRequest is used to query the count (and state eventually)
+// of the scheduler workers running in a Nomad server agent.
+// This endpoint can also be used to update the count of running workers for a
+// given agent.
+func (s *HTTPServer) AgentSchedulerWorkerConfigRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	if s.agent.Server() == nil {
+		return nil, CodedError(http.StatusBadRequest, ErrServerOnly)
+	}
+	switch req.Method {
+	case "PUT", "POST":
+		return s.UpdateScheduleWorkersConfig(resp, req)
+	case "GET":
+		return s.GetScheduleWorkersConfig(resp, req)
```

Comment:
We're probably not consistent about this across the code base, but good to use the constants for new code at least:

```suggestion
	switch req.Method {
	case http.MethodPut, http.MethodPost:
		return s.UpdateScheduleWorkersConfig(resp, req)
	case http.MethodGet:
		return s.GetScheduleWorkersConfig(resp, req)
```